### PR TITLE
Dynamically load the workflow schema

### PR DESCRIPTION
**Pull request by EmilyGavrilenko**
# Description

Create an endpoint to dynamically load the workflow schema to be used in the JSON editor syntax checker

Todo: 
- Create tests

## Type of change
-   [X] Bug fix (non-breaking change which fixes an issue)

## How has this change been tested, please provide a testcase or example of how you tested the change?
Locally

## Any specific deployment considerations

For example, documentation changes, usability, usage/costs, secrets, etc.

## Docs

-   [ ] Docs updated? What were the changes:

### CODE REVIEW

File: inference/core/interfaces/http/http_api.py [hunk 1]
```diff
@@ -943,6 +945,17 @@ async def describe_workflows_blocks(
                     dynamic_blocks_definitions=dynamic_blocks_definitions
                 )
 
+            @app.get(
+                "/workflows/blocks/schema",
```
Comment (hansent) [inline inference/core/interfaces/http/http_api.py:949]: should the path be `/workflows/spec/schema` or similar since the schema is for the spec rather than blocks per se

Comment (hansent) [reply inference/core/interfaces/http/http_api.py:949]: also need to add the new endpoint to the API gateway for this to work with hosted workflows.

This needs to be done in roboflow-infra, I think in this file:  https://github.com/roboflow/roboflow-infra/blob/1ab773c66633dbc0311443200877d5768fd0ef61/aws/serverless/modules/containerlambda/main.tf#L110

Comment (PawelPeczek-Roboflow) [reply inference/core/interfaces/http/http_api.py:949]: We call the JSON workflow definition, so maybe `GET /workflows/definition/schema`

File: inference/core/workflows/execution_engine/compiler/syntactic_parser.py [hunk 2]
```diff
@@ -51,9 +52,15 @@ def build_workflow_definition_entity(
     block_manifest_types_union = Union[steps_manifests]
     block_type = Annotated[block_manifest_types_union, Field(discriminator="type")]
     return create_model(
-        "WorkflowSpecificationV1",
+        "WorkflowSpecificationV2",
```
Comment (PawelPeczek-Roboflow) [inline inference/core/workflows/execution_engine/compiler/syntactic_parser.py:55]: Is that change needed

**Review by PawelPeczek-Roboflow — COMMENTED** [review]
LGTM, but let's agree on the endpoint name and clarify change in the model name

File: inference/core/interfaces/http/http_api.py (hunk 1 above)
Comment (EmilyGavrilenko) [reply inference/core/interfaces/http/http_api.py]: Renaming to `/workflows/definition/schema`

File: inference/core/workflows/execution_engine/compiler/syntactic_parser.py (hunk 2 above)
Comment (EmilyGavrilenko) [reply inference/core/workflows/execution_engine/compiler/syntactic_parser.py]: initially added to make sure it was fetching the new version on the frontend, will revert back
